### ISSUE TITLE
Option to skip Addon dependencies

### ISSUE THREAD

**Issue by benoror** [rwjblue/ember-template-lint#119]
One addon is offending with `bare-strings`, the error is caught during our main app build:

```
bare-strings: Non-translated string used (modules/ember-formly/templates/components/ember-formly-component-unknown @ L1:C5):
`Ember Formly: Unknown Component`
===== 1 Template Linting Error
```

Is there a way to skip linting addon dependencies?

**Comment by benoror** [comment]
I tried whitelisting the whole phrase but it won't work either:

.template-lintrc.js
```
/* jshint node:true */
'use strict';

module.exports = {
  extends: 'recommended',

  rules: {
    'bare-strings': [
      '|', ':', '-', ',', '*', '.',
      'Ember Formly: Unknown Component'
    ],
    'block-indentation': false,
    'img-alt-attributes': true,
    'link-rel-noopener': true,
    'invalid-interactive': true,
    'style-concatenation': true
  }
};
```

**Comment by benoror** [comment]
As a temp solution this did the trick:

```javascript
  pending: ['modules/ember-formly/templates/components/ember-formly-component-unknown']
```

Still seems odd to me that dependencies are being linted

Great addon btw, kudos!

**Comment by rwjblue** [comment]
In general ember-cli-template-lint does not lint addon modules, but occasionally addon authors forget to remove `isDevelopingAddon` overrides when publishing.  When that happens the addon's templates would get linted as you are seeing here...

**Comment by rwjblue** [comment]
I'm going to close this as it is likely to be an incorrectly published addon, but if I'm mistaken I'm happy to reopen...

**Comment by benoror** [comment]
Thanks Robert! fyi the addon's `isDevelopingAddon` is not set: https://github.com/benoror/ember-formly/blob/master/index.js

And the error is showing in a CI, so no chances of `npm link` issues.

Anyway, I'll keep an eye on it and see if something else pops up

Cheers!

**Comment by rwjblue** [comment]
@benoror - Well, it was set when it was published.  Please double check my steps though to make sure I haven't mucked something up:

```
curl -O https://registry.npmjs.org/ember-formly/-/ember-formly-0.0.5.tgz
tar -xzvf ember-formly-0.0.5.tgz
cat packages/index.js
```

The output I get from the above is:

```js
/* jshint node: true */
'use strict';

module.exports = {
  name: 'ember-formly',
  isDevelopingAddon: function() {
    return true;
  }
};
```

**Comment by rwjblue** [comment]
One thing that is a definite "gotcha" with NPM packages is that `npm pack` and `npm publish` use your local working copy to build the tarball sent up to the registry, this means that any dirty files are packed as they were locally during publishing (and they can differ from what is checked into source control).

**Comment by benoror** [comment]
@rwjblue gotcha, that's the issue indeed, didn't know that about `npm publish`, thx for the tip!
